statik/utils.py

```diff
@@ -385,5 +387,30 @@ def find_duplicates_in_array(array):
 
     return duplicates
 
+
 def camel_to_snake(camel):
     return '_'.join(re.findall(r'[A-Z][a-z]*', camel))
+
+
+def validate_jinja_template(template, ctx):
+    """Checks that variables used in templates are actually available to be used in context.
+    Logs a warning if a variable used in the template is not found in the context.
+
+    Args:
+        template: Path to the template to check
+        ctx: dictionary of the variables in context
+    """
+    tag_re = re.compile(r'(<!--.*?-->|<[^>]*>)')
+    with open(template) as file:
+        stripped_template = tag_re.sub('', file.read())
+        try:
+            result = to_json_schema(infer(stripped_template))
+            for item in result['required']:
+                if item not in list(ctx.keys()):
+                    logger.warning("'%s' was used in a template, "
+                                   "but it's not available to be used. Template: %s"
+                                   % (item, template))
+        except Exception as e:
+            logger.warning(
+                "Template validation failed (likely an issue with jinja2schema): %s: %s",
```

Comment:
move the `except` so that it only catches errors in one line `result = to_json_schema(infer(stripped_template))` and then change the warning to explicitly indicate the problem is within `jinja2schema`.  Something like "Template inspection using jinja2schema failed: .."

Usually, only do one thing within try/except blocks.  Then add another exception block to catch the next type of error.